projects/llm-d/testing/test_llmd.py

```
@@ -1216,6 +1220,43 @@ def capture_llm_inference_service_state(llmisvc_name, namespace):
+
+        # Check if router scheduler exists
+        has_scheduler = 'router' in isvc_data.get('spec', {}) and 'scheduler' in isvc_data['spec'].get('router', {})
+
+        if has_scheduler:
+            logging.info("Scheduler found in ISVC spec - will capture scheduler metrics")
+        else:
+            logging.info("No scheduler found in ISVC spec - will skip scheduler metrics")
+
+        return has_scheduler
+
+    except Exception as e:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)